kits/daisy/src/input/input_manager_mobile_io.cpp

```diff
@@ -0,0 +1,137 @@
+#include "input_manager_mobile_io.hpp"
+#include "lookup.hpp"
+#include "xml_util/pugixml.hpp"
+#include "group_feedback.hpp"
+
+#include <iostream> // Note: for debugging output.
+
+namespace hebi {
+namespace input {
+
+InputManagerMobileIO::InputManagerMobileIO()
+{
+  if (!reset())
+    std::cout << "Could not find joystick input board!" << std::endl;
+}
+
+bool InputManagerMobileIO::reset()
+{
+  // Cleanup if we already have one!
+  if (group_)
+    group_.reset();
+
+  // Look for joystick (IO board)
+  hebi::Lookup lookup;
+  long timeout_ms = 2000; // use a 2 second timeout
+  group_ = lookup.getGroupFromNames({"HEBI"}, {"Mobile IO"}, timeout_ms);
+  if (!group_)
+    return false;
+
+  group_->addFeedbackHandler([this] (const GroupFeedback& fbk)
+  {
+    // TODO: use 'std::atomic' variables for state here? Add mutex?  Does this
+    // slow things down due to lock contention?
```

Comment:
There are a lot of other things which need to be cleaned up which will result in better performance than lock contention here

Reply:
Again, a comment that doesn't really need any resolution ATM